packages/babel-parser/src/parser/location.js

```diff
@@ -26,33 +208,40 @@ export default class LocationParser extends CommentsParser {
     return loc;
   }
 
-  raise(
+  raise(pos: number, errorTemplate: string, ...params: any): Error | empty {
+    return this.raiseWithData(pos, undefined, errorTemplate, ...params);
+  }
+
+  raiseWithData(
     pos: number,
-    message: string,
-    {
-      missingPluginNames,
-      code,
-    }: {
-      missingPluginNames?: Array<string>,
+    data?: {
+      missingPlugin?: Array<string>,
       code?: string,
-    } = {},
+    },
+    errorTemplate: string,
+    ...params: any
   ): Error | empty {
     const loc = this.getLocationForPosition(pos);
-
-    message += ` (${loc.line}:${loc.column})`;
-    // $FlowIgnore
-    const err: SyntaxError & { pos: number, loc: Position } = new SyntaxError(
+    const message =
+      errorTemplate.replace(/%(\d+)/g, (_, i: number) => params[i]) +
+      ` (${loc.line}:${loc.column})`;
+    return this._raise(
+      Object.assign(
+        {},
+        {
+          loc,
+          pos,
+        },
+        data,
+      ),
```

Comment:
The first empty object isn't needed.

```suggestion
      Object.assign({ loc, pos }, data),
```

Reply:
It was required by Flow otherwise `{ loc, pos }` is a sealed object. https://also.github.io/notes/2016/03/13/flow-object-assign.html

If we are okay with that I can `FlowIgnore` it lol.

Reply:
Maybe this?

```suggestion
      Object.assign(({ loc, pos }: Object), data),
```